### PR TITLE
Double Chain Swap Example

### PR DESCRIPTION
**Pull request by k3ithl1m**
a simple example for a double chain swap

Still in progress

### CODE REVIEW

File: doubleChainSwap.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+val sndOut = Box(value = nanoergsInErg / 2,
+                token = (fstToken -> fstTokenAmount),
+                register = (R4 -> sndOrderTransactionSigned.outputs(0).id),
+                script = contract(sndHodler.wallet.getAddress.pubKey))
+
+// Create the swap transaction
+val swapTransaction = Transaction(
+    inputs = List(sndOrderTransactionSigned.outputs(0), fstOrderTransactionSigned.outputs(0)),
+    outputs = List(fstOut, sndOut),
+    sendChangeTo = fstHodler.wallet.getAddress,
```
Comment (k3ithl1m) [inline doubleChainSwap.md:180]: In this situation, the change is sent to fstHodler. What would instead be the recommended situation? Should the change get burned? Can the change be split and sent to both parties?

File: doubleChainSwap.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+// Define a swapper sndHodler (with a wallet)
+val sndHodler = blockchainSim.newParty("SND Hodler")
+
+///////////////////////////////////////////////////////////////////////////////////
+// Double-chain Swap Contracts - Swap Contract                                    //
+///////////////////////////////////////////////////////////////////////////////////
+
+val swapScript = s"""
+  {
+    val defined = OUTPUTS(0).R2[Coll[(Coll[Byte], Long)]].isDefined && OUTPUTS(0).R4[Coll[Byte]].isDefined
```
Comment (kushti) [inline doubleChainSwap.md:56]: R2 is always defined (likely you wanted to check for !...isEmpty ?

File: doubleChainSwap.md (hunk 1 above)
Comment (kushti) [reply doubleChainSwap.md:171]: Different options are possible I guess, you can't burn ERG though. sending to one party is okay I suppose (especially in an example)

Comment (k3ithl1m) [reply doubleChainSwap.md:211]: I reduced the transaction fee to minimum and it solved the issue.

File: doubleChainSwap.md (hunk 2 above)
Comment (k3ithl1m) [reply doubleChainSwap.md]: Figured that it is unnecessary then.

**Review by kushti — APPROVED** [review]

**Comment by kushti** [comment]
@k3ithl1m accepted, please contact me over Discord to get a reward

**Pull request merged**